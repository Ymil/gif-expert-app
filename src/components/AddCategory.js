import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {

    const [inputValue, setInputValue] = useState("");

    const handlerInputChange = (e) => {    
        setInputValue(e.target.value);
    }

    const handlerSumit = (e) => {
        if(inputValue.trim().length > 2){
            setCategories(categories => [inputValue, ...categories]);
            setInputValue('')
        }
        
        
        e.preventDefault()
    }

    return (
        <form onSubmit={ handlerSumit }>         
            <input 
            type='text'
            value={inputValue}
            onChange={ handlerInputChange }
            placeholder='Ingrese la categoria a buscar'
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
