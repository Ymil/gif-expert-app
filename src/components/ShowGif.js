import React, { useState } from 'react'

export const ShowGif = ({id, title, url}) => {
    
    const [state, setstate] = useState(false);

    return (
        <div className='card animate__animated animate__pulse'>
            { !state && <p className="animate__animated animate__flash animate__infinite">Cargando...</p>}
            <img src={url} onLoad={() => setstate(true)}/>
            <p>{title.trim().length == 0 ? "Sin titulo" : title}</p>
        </div>
    )
}
