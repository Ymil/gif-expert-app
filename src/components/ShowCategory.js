import PropTypes from 'prop-types'
import { ShowGif } from './ShowGif'
import { useFetchGifs } from './hooks/useFetchGifs'
import "animate.css"
export const ShowCategory = ({category}) => {
    const {gifs, loading} = useFetchGifs(category);
    
    return (
        <>
            <h2>{category}</h2>

            {
                loading && <p className="animate__animated animate__flash">Cargando...</p>
            }
            <div className="card-grid">
                { gifs.map(
                    (gif) => <ShowGif {...gif} key={gif.id} />
                )}
            </div>
        </>
    )
}

ShowCategory.propTypes = {
    category: PropTypes.string.isRequired
}


