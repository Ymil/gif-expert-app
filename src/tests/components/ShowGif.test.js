import { shallow } from "enzyme"
import { ShowGif } from "../../components/ShowGif"

describe('Progando showgif', () => {
    let title = "Imagen de deportes";
    const url = "https://media1.giphy.com/media/qYxZiiuxQIiexrKqj3/giphy.gif?cid=3a8a57599peodb6t6zyy2238q7w80r88zh1oweg5udlwg4du&rid=giphy.gif&ct=g";
    const wrapper = shallow(
        <ShowGif title={title} url={url}/>
    );
    test('Probando carga general', () => {        
        expect(wrapper).toMatchSnapshot()
    });

    test('Debe tener el parrafo title', () => {
        const p = wrapper.find('.title');
        expect(p.text()).toBe(title);
    });

    test('Probando titulo vacio', () => {
        let title = "";
        const wrapper = shallow(
            <ShowGif title={title} url={url}/>
        );
        const p = wrapper.find('.title');
        expect(p.text()).toBe("Sin titulo");
    });
    
    test('la imagen debe tener la misma url que la que se paso por props', () => {
        const img = wrapper.find("img");
        expect(img.prop('src')).toBe(url)
    });

    test('verificando clases', () => {
        const {className} = wrapper.props('className');        
        expect(className.includes('animate__pulse')).toBe(true)
    })
    

})
