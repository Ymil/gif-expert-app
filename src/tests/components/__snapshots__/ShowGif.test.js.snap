// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progando showgif Probando carga general 1`] = `
<div
  className="card animate__animated animate__pulse"
>
  <p
    className="animate__animated animate__flash animate__infinite"
  >
    Cargando...
  </p>
  <img
    alt="Imagen de deportes"
    onLoad={[Function]}
    src="https://media1.giphy.com/media/qYxZiiuxQIiexrKqj3/giphy.gif?cid=3a8a57599peodb6t6zyy2238q7w80r88zh1oweg5udlwg4du&rid=giphy.gif&ct=g"
  />
  <p
    className="title"
  >
    Imagen de deportes
  </p>
</div>
`;
